### ISSUE TITLE
[Feature] Add validation for Definition Object

### ISSUE THREAD

**Issue by wonderflow** [crossplane/oam-kubernetes-runtime#247]
For traitDefinition, the value of applyTo could have many formats:

* format by  `group/version.Kind`  => core.oam.dev/v1alpha2.PodspecWorkload
* directly use `workloadDefinition Name` => webservice
* directly use `CRD name` => deployments.apps
* use `group name` to represent apply to the whole group => `core.oam.dev`

We need to add validation for it and future needs.

**Comment by LeoLiuYan** [comment]
Now, when apply traits, the workloads does not add to traitDefinition's AppliesToWorkloads field like the scopes apply action. So, the applyTo field maybe not use in this moment.  
The associated code maybe here:

**Comment by LeoLiuYan** [comment]
Now, when apply traits, the workloads does not add to traitDefinition's AppliesToWorkloads field like the scopes apply do. So, the applyTo field maybe not use in this moment.

The associated code maybe here: 
https://github.com/crossplane/oam-kubernetes-runtime/blob/master/pkg/controller/v1alpha2/applicationconfiguration/apply.go#L104
https://github.com/crossplane/oam-kubernetes-runtime/blob/master/pkg/controller/v1alpha2/applicationconfiguration/apply.go#L121

**Comment by wonderflow** [comment]
Yes, checking applyTo fields should also add to this validation. @LeoLiuYan

**Comment by LeoLiuYan** [comment]
Could this feature assign to me? @wonderflow

**Comment by wonderflow** [comment]
Sure, we have already setup validation webhook framework https://github.com/crossplane/oam-kubernetes-runtime/tree/master/pkg/webhook/v1alpha2 

You could add it there. @LeoLiuYan
